microprofile/reactive-messaging-smallrye/kafka/src/main/java/org/wildfly/microprofile/reactive/messaging/config/kafka/ssl/context/ReactiveMessagingSslConfigProcessor.java

```diff
@@ -0,0 +1,128 @@
+
+    private static final String SSL_ENGINE_FACTORY_CLASS = "ssl.engine.factory.class";
+
+    private static final AttachmentKey<Object> DEPLOYMENT_ATTACHEMENT_KEY = AttachmentKey.create(Object.class);
+
+    @Override
+    public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitProcessingException {
+        DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
+        if (!isReactiveMessagingDeployment(deploymentUnit)) {
+            return;
+        }
+        Module module = deploymentUnit.getAttachment(Attachments.MODULE);
+
+        Config config = ConfigProviderResolver.instance().getConfig(module.getClassLoader());
+
+        String sslEngineFactoryClassName = getSSLEngineFactoryClassName();
+        if (sslEngineFactoryClassName != null) {
+            Map<String, String> addedProperties = new HashMap<>();
+
+            Set<ServiceName> mscDependencies = new HashSet<>();
+            for (String propertyName : config.getPropertyNames()) {
+                if (propertyName.endsWith(SSL_CONTEXT_PROPERTY_SUFFIX)) {
+                    String propertyValue = config.getValue(propertyName, String.class);
+                    String prefix = null;
+                    if (propertyName.equals(RM_KAFKA_GLOBAL_PROPERTY_PREFIX + SSL_CONTEXT_PROPERTY_SUFFIX)) {
+                        prefix = RM_KAFKA_GLOBAL_PROPERTY_PREFIX;
+                    } else if (propertyName.startsWith(RM_INCOMING_PROPERTY_PREFIX) ||
+                            propertyName.startsWith(RM_OUTGOING_PROPERTY_PREFIX)) {
+                        prefix = propertyName.substring(0, propertyName.indexOf(SSL_CONTEXT_PROPERTY_SUFFIX));
+                    }
+                    if (prefix != null) {
+                        LOGGER.foundPropertyUsingElytronClientSSLContext(propertyName, propertyValue);
+                        SSLContext ctx = ElytronSSLContextRegistryService.getInstalledSSLContext(propertyValue);
+                        if (ctx == null) {
+                            throw LOGGER.noElytronClientSSLContext(propertyValue);
+                        }
+                        mscDependencies.add(ElytronSSLContextRegistryService.getSSLContextName(propertyValue));
+                        addedProperties.put(prefix + SSL_ENGINE_FACTORY_CLASS, sslEngineFactoryClassName);
+                    }
+                }
+            }
+            if (addedProperties.size() > 0) {
+                ReactiveMessagingConfigSource.addProperties(config, addedProperties);
+                for (ServiceName svcName : mscDependencies) {
+                    phaseContext.addDeploymentDependency(svcName, DEPLOYMENT_ATTACHEMENT_KEY);
+                }
+            }
+        }
+    }
+
+    private String getSSLEngineFactoryClassName() {
+        try {
+            Class<?> clazz = WildFlyKafkaSSLEngineFactory.class;
+            return clazz.getName();
+        } catch (Throwable throwable) {
+            return null;
```

Comment:
A comment explaining why returning null is ok would be good.

I haven't verified but my assumption is the reason is the module that provides WildFlyKafkaSSLEngineFactory may not be provisioned, and that's ok.

Reply:
I think this is a hangover from before I put this class into the same module as WildFlyKafkaSSLEngineFactory - so the load should always succeed. Will fix